xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -349,24 +448,42 @@ public static String escapeElementContent(Object content)
      * Unescape encoded special XML characters. Only &gt;, &lt; &amp;, ", ' and { are unescaped, since they are the only
      * ones that affect the resulting markup.
      *
-     * @param content the text to decode, may be {@code null}
+     * @param content the text to decode, may be {@code null}. The content is converted to {@code String} using
+     * {@link String#valueOf(Object)} before escaping.
      * @return unescaped content, {@code null} if {@code null} input
+     * @deprecated since 12.8RC1, use {@link #unescape(String)} instead
      */
+    @Deprecated
     public static String unescape(Object content)
```

Comment:
I think it should be easy in your IDE to find all existing calls to `unescape(Object)` in our code base and replace them with calls to `unescape(String)`. This would allow you/us to move the deprecated method to legacy (see "Second Step" in https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HBackwardCompatibility).

WDYT?

Reply:
Doing it. 
Also:
1. I forgot to add javadoc `@since` for the new methods (fixing)
2. Should I add an `@Unstable` annotation on the new methods?

Reply:
> Should I add an @Unstable annotation on the new methods?

Yes

Reply:
> Doing it.

Cool, thanks!

Reply:
Deprecated and since added.

Reply:
Call to the deprecated methods updated in https://github.com/xwiki/xwiki-platform/pull/1384